units/en/unit2/smolagents/tools.mdx

```diff
@@ -0,0 +1,267 @@
+
+To do this, he can use an agent that generates superhero-themed party ideas based on a given category. This way, Alfred can find the perfect party theme to wow his guests.
+
+```python
+from smolagents import Tool, CodeAgent, HfApiModel
+
+class SuperheroPartyThemeTool(Tool):
+    name = "superhero_party_theme_generator"
+    description = """
+    This tool suggests creative superhero-themed party ideas based on a category.
+    It returns a unique party theme idea."""
+    
+    inputs = {
+        "category": {
+            "type": "string",
+            "description": "The type of superhero party (e.g., 'classic heroes', 'villain masquerade', 'futuristic Gotham').",
+        }
+    }
+    
+    output_type = "string"
+
+    def forward(self, category: str):
+        themes = {
+            "classic heroes": "Justice League Gala: Guests come dressed as their favorite DC heroes with themed cocktails like 'The Kryptonite Punch'.",
+            "villain masquerade": "Gotham Rogues' Ball: A mysterious masquerade where guests dress as classic Batman villains.",
+            "futuristic Gotham": "Neo-Gotham Night: A cyberpunk-style party inspired by Batman Beyond, with neon decorations and futuristic gadgets."
+        }
+        
+        return themes.get(category.lower(), "Themed party idea not found. Try 'classic heroes', 'villain masquerade', or 'futuristic Gotham'.")
+
+# Instantiate the tool
+party_theme_tool = SuperheroPartyThemeTool()
+agent = CodeAgent(tools=[party_theme_tool], model=HfApiModel())
+
+# Run the agent to generate a party theme idea
+result = agent.run(
+    "What would be a good superhero party idea for a 'villain masquerade' theme?"
+)
+
+print(result)  # Output: "Gotham Rogues' Ball: A mysterious masquerade where guests dress as classic Batman villains."
+```
+
+With this tool, Alfred will be the ultimate super host, impressing his guests with a superhero-themed party they won't forget! 🦸‍♂️🦸‍♀️
+
+## Default Toolbox  
+
+`smolagents` comes with a set of pre-built tools that can be directly injected into your agent. The [default toolbox](https://huggingface.co/docs/smolagents/guided_tour?build-a-tool=Decorate+a+function+with+%40tool#default-toolbox) includes:  
+
+- **PythonInterpreterTool**  
+- **FinalAnswerTool**  
+- **UserInputTool**  
+- **DuckDuckGoSearchTool**  
+- **GoogleSearchTool**  
+- **VisitWebpageTool**  
+
+Alfred could use various tools to ensure a flawless party at Wayne Manor:
+
+- First, he could use the `DuckDuckGoSearchTool` to find creative superhero-themed party ideas. 
+
+- For catering, he'd rely on the `GoogleSearchTool` to find the highest-rated services in Gotham. 
+
+- To manage seating arrangements, Alfred could run calculations with the `PythonInterpreterTool`. 
+
+- Once everything is gathered, he'd compile the plan using the `FinalAnswerTool`. 
+
+- Lastly, the `VisitWebpageTool` would keep him updated on any changes, ensuring a smooth event. 
```

Comment:
Maybe skip this one? The list up to this point ends naturally with compiling the final plan.